src/Components/Facility/FacilityHome.tsx

```diff
@@ -80,76 +73,99 @@ export const FacilityHome = (props: any) => {
+  });
+
+  const {
+    res: doctorRes,
+    data: doctorFetchData,
+    refetch: doctorFetch,
+  } = useQuery(routes.listDoctor, {
+    pathParams: { facilityId: facilityId },
+  });
+
+  useEffect(() => {
+    const fetchData = async () => {
       setIsLoading(true);
-      const facilityRes = await dispatch(getPermittedFacility(facilityId));
-      if (facilityRes) {
-        const [capacityRes, doctorRes, triageRes] = await Promise.all([
-          dispatch(listCapacity({}, { facilityId })),
-          dispatch(listDoctor({}, { facilityId })),
-          dispatch(getTriageInfo({ facilityId })),
-        ]);
-        if (!status.aborted) {
+      if (permittedFacilityRes?.ok) {
+        capacityFetch();
+        doctorFetch();
+        const { res: triageRes, data: triageData } = await request(
+          routes.getTriage,
+          {
+            pathParams: { facilityId },
+          }
+        );
+
+        if (permittedFacilityRes) {
           setIsLoading(false);
-          if (!facilityRes.data) {
+          if (!permittedFacilityData) {
             Notification.Error({
               msg: "Something went wrong..!",
             });
           } else {
-            setFacilityData(facilityRes.data);
-            if (capacityRes && capacityRes.data) {
-              setCapacityData(capacityRes.data.results);
+            setFacilityData(permittedFacilityData);
+            if (capacityRes?.ok && capacityFetchData) {
+              setCapacityData(capacityFetchData.results);
             }
-            if (doctorRes && doctorRes.data) {
-              setDoctorData(doctorRes.data.results);
+            if (doctorRes?.ok && doctorFetchData) {
+              setDoctorData(doctorFetchData.results);
               // calculating total doctors count
               let totalCount = 0;
-              doctorRes.data.results.map((doctor: DoctorModal) => {
+              [doctorFetchData].map((doctor: DoctorModal) => {
                 if (doctor.count) {
                   totalCount += doctor.count;
                 }
               });
               setTotalDoctors(totalCount);
             }
             if (
-              triageRes &&
-              triageRes.data &&
-              triageRes.data.results &&
-              triageRes.data.results.length
+              triageRes?.ok &&
+              triageData &&
+              triageData.results &&
+              triageData.results.length
             ) {
-              setPatientStatsData(triageRes.data.results);
+              setPatientStatsData(triageData.results);
             }
           }
         }
       } else {
         navigate("/not-found");
         setIsLoading(false);
       }
-    },
-    [dispatch, facilityId]
-  );
-
-  useAbortableEffect(
-    (status: statusType) => {
-      fetchData(status);
-    },
-    [dispatch, fetchData]
-  );
+    };
+    if (permittedFacilityRes?.ok) fetchData();
+  }, [facilityId, permittedFacilityRes, permittedFacilityData]);
```

Comment:
Delete this entire useEffect. use onResponse of the above related useQueries instead.